tt_metal/llrt/tt_cluster.cpp

```diff
@@ -37,6 +37,8 @@ Cluster::Cluster() {
 
     this->detect_arch_and_target();
 
+    tt_metal::hal.initialize(arch_);
```

Comment:
I think there is an extra hal.initialize call in `impl/device.cpp`, could you remove that from there?

Reply:
I think it is already removed.